src/appleseed/foundation/math/population.h

```diff
@@ -134,12 +134,12 @@ inline void Population<T>::insert(const ValueType& val)
     if (m_max < val)
         m_max = val;
 
-    // Update the size of the population.
-    ++m_size;
-
     // Compute the residual value.
     const double double_val = static_cast<double>(val);
-    const double residual = double_val - m_mean;
+    const double residual = (double_val - m_mean) * amount;
```

Comment:
I haven't checked the math. Are you sure about this formula?

Reply:
I wrote a test or two. Do you want me to write more test ?